homeassistant/components/konnected.py

```diff
@@ -0,0 +1,298 @@
+"""
+Support for Konnected devices.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/konnected/
+"""
+import asyncio
+import logging
+import voluptuous as vol
+
+from aiohttp.hdrs import AUTHORIZATION
+from aiohttp.web import Request, Response  # NOQA
+
+from homeassistant.components.binary_sensor import DEVICE_CLASSES_SCHEMA
+from homeassistant.components.discovery import SERVICE_KONNECTED
+from homeassistant.components.http import HomeAssistantView
+from homeassistant.const import (
+    HTTP_BAD_REQUEST, HTTP_INTERNAL_SERVER_ERROR, HTTP_UNAUTHORIZED,
+    CONF_DEVICES, CONF_SENSORS, CONF_SWITCHES, CONF_HOST, CONF_PORT,
+    CONF_ID, CONF_NAME, CONF_TYPE, CONF_PIN, CONF_ZONE, ATTR_STATE)
+from homeassistant.helpers import discovery
+from homeassistant.helpers import config_validation
+
+_LOGGER = logging.getLogger(__name__)
+
+REQUIREMENTS = ['konnected==0.1.2']
+
+DOMAIN = 'konnected'
+
+PIN_TO_ZONE = {1: 1, 2: 2, 5: 3, 6: 4, 7: 5, 8: 'out', 9: 6}
+ZONE_TO_PIN = {zone: pin for pin, zone in PIN_TO_ZONE.items()}
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema({
+            vol.Required('auth_token'): config_validation.string,
+            vol.Required(CONF_DEVICES): [{
+                vol.Required(CONF_ID, default=''): config_validation.string,
+                vol.Optional(CONF_SENSORS): [{
+                    vol.Exclusive(CONF_PIN, 's_pin'): vol.Any(*PIN_TO_ZONE),
+                    vol.Exclusive(CONF_ZONE, 's_pin'): vol.Any(*ZONE_TO_PIN),
+                    vol.Required(CONF_TYPE, default='motion'):
+                        DEVICE_CLASSES_SCHEMA,
+                    vol.Optional(CONF_NAME): config_validation.string,
+                }],
+                vol.Optional(CONF_SWITCHES): [{
+                    vol.Exclusive(CONF_PIN, 'a_pin'): vol.Any(*PIN_TO_ZONE),
+                    vol.Exclusive(CONF_ZONE, 'a_pin'): vol.Any(*ZONE_TO_PIN),
+                    vol.Optional(CONF_NAME): config_validation.string,
+                    vol.Required('activation', default='high'):
+                        vol.All(vol.Lower, vol.Any('high', 'low'))
+                }],
+            }],
+        }),
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+DEPENDENCIES = ['http']
+
+ENDPOINT_ROOT = '/api/konnected'
+UPDATE_ENDPOINT = (
+    ENDPOINT_ROOT +
+    r'/device/{device_id:[a-zA-Z0-9]+}/{pin_num:[0-9]}/{state:[01]}')
+
+
+@asyncio.coroutine
+def async_setup(hass, config):
+    """Set up the Konnected platform."""
+    cfg = config.get(DOMAIN)
+    if cfg is None:
+        cfg = {}
+
+    auth_token = cfg.get('auth_token')
+    if DOMAIN not in hass.data:
+        hass.data[DOMAIN] = {'auth_token': auth_token}
+
+    @asyncio.coroutine
```

Comment:
Please use the new `async`/`await` syntax if you can. We have Python 3.5 now and I've seen a few PRs getting comments like this one. See http://stackabuse.com/python-async-await-tutorial/ for a simple tutorial.

Reply:
Ok, I will read up on it. I'm pretty new to Python so I'm sure that I just modeled this after some other code that maybe wasn't up date.